### ISSUE TITLE
pkg_resources.DistributionNotFound: The 'umi-tools==0.2.3' distribution was not found and is required by the application

### ISSUE THREAD

**Issue by ChristianRohde** [CGATOxford/UMI-tools#584]
Hi, 

I installed **umi_tools** today with _conda install -c bioconda -c conda-forge umi_tools_. The installation took quite some time but finally it pulled a version which could be installed. Unfortunately the installation gave me the following error: 

> Traceback (most recent call last):
>   File "/path/to/conda/bin/umi_tools", line 5, in <module>
>     from pkg_resources import load_entry_point
>   File "/path/to/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 3260, in <module>
>     def _initialize_master_working_set():
>   File "/path/to/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 3234, in _call_aside
>     f(*args, **kwargs)
>   File "/path/to/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 3272, in _initialize_master_working_set
>     working_set = WorkingSet._build_master()
>   File "/path/to/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 581, in _build_master
>     ws.require(__requires__)
>   File "/path/to/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 909, in require
>     needed = self.resolve(parse_requirements(requirements))
>   File "/path/to/conda/lib/python3.9/site-packages/pkg_resources/__init__.py", line 795, in resolve
>     raise DistributionNotFound(req, requirers)
> pkg_resources.DistributionNotFound: The 'umi-tools==0.2.3' distribution was not found and is required by the application

Interestingly I installed the following:
python                    3.9.13          h9a8a25e_0_cpython    conda-forge
umi_tools                 0.2.3                         1    bioconda

Next I tried a conda environment using python=3.8. Same problem. Finally I ended up with a conda environment using python=3.6 which worked out of the box:
python                    3.6.15          hb7a2778_0_cpython    conda-forge
umi_tools                 1.1.4            py36h91eb985_0    bioconda

I am not sure if this is the most recent and recommend version of umi_tools. Maybe you can solve the problem with dependencies for python > 3.6.

Best,
Christian

**Comment by IanSudbery** [comment]
I think there is more than one thing going on here. 

Firstly, I am able to build and install an env with umi_tools 1.1.4 (which is the lastest release, released only yesterday) and python 3.8 or python 3.10 with

```
$mamba create python=3.8[or python=3.10] umi_tools=1.1.4 -c bioconda -c conda-forge
```

If I don't specify the version of umi_tools, for some reason I am getting 1.1.0, which, while not the most recent, should work (and much better than 0.2.3, which is I think, nearly 6 years old, and is almost certainly completely broken). 

Thus, I guess that your problems with python 3.8 is probably due to trying to install into an already build env that has some other incompatible package installed. 

Python 3.9 however, is different. For some reason bioconda is building umi_tools 1.1.4 for python 3.6, 3.7, 3.8 and 3.10, but not 3.9. 

We don't have any direct control over bioconda, but I am talking to the bioconda people to see if we can work out what is happening.

**Comment by ChristianRohde** [comment]
great, you are totally right: once I started with a fresh conda installation I could successfully install umi_tools version 1.1.4 with python 3.10.8!

On the other hand also here it took quite long and it failed 2 times. Finally python got downgraded from 3.10.9-he550d4f_0_cpython --> 3.10.8-h257c98d_0_cpython, please find more details below. But well, I am still happy with this.

Best, 
Christian

wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
bash miniconda.sh -b -p $( ws_find WS)/conda
conda update -n base conda
conda config --add channels defaults                                                                                                   
conda config --add channels bioconda                                                                                                   
conda config --add channels conda-forge                                                                                                
conda install -c bioconda umi_tools                                                                                                    
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.                   
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.                
Collecting package metadata (repodata.json): done                
Solving environment: done

python                    3.10.8          h257c98d_0_cpython    conda-forge
umi_tools                 1.1.4           py310h1425a21_0    bioconda

**Comment by IanSudbery** [comment]
Probably the reason for the long solve is that by default all you will get the lastest version of `openssl` (3.0.x) installed with you make a python env. However, `umi_tools` requires pysam. And pysam requires `openssl<1.1.2`

BTW, its generally though of as bad practice to install things into the base environment, which is what you appear to be doing here. You should not need an entirely fresh conda install to build a fresh environment.

**Comment by IanSudbery** [comment]
bioconda has now fixed the lack of a python 3.9 build with PR https://github.com/bioconda/bioconda-recipes/pull/39728.

Thus, I'm going to close this. Please reopen if you continue to have difficulty.
